docs/usage/rtk-query/cached-data.md

```diff
@@ -0,0 +1,516 @@
+```
+
+By declaring these tags as what can possibly be provided to the cache, it enables control for individual endpoints to claim whether they affect specific portions of the cache or not, in conjunction with `providesTags` and `invalidatesTags` on individual endpoints.
+
+### Providing cache data
+
+Each individual `query` endpoint can `provide` particular tags to the cache. Doing so enables a relationship between cached data from one or more `query` endpoints and the behaviour of one or more `mutation` endpoints.
+
+The `providesTags` property on a `query` endpoint is used for this purpose.
+
+The example below declares that the `getPosts` `query` endpoint `provides` the `'Post'` tag to the cache, using the `providesTags` property for a `query` endpoint.
+
+```js title="Example of providing tags to the cache"
+const api = createApi({
+  baseQuery,
+  tagTypes: ['Post', 'User'],
+  endpoints: (build) => ({
+    getPosts: build.query({
+      query: () => '/posts',
+      // highlight-start
+      providesTags: ['Post'],
+      // highlight-end
+    }),
+    getUsers: build.query({
+      query: () => '/users',
+      // highlight-start
+      providesTags: ['User'],
+      // highlight-end
+    }),
+    addPost: build.mutation({
+      query: (body) => ({
+        url: 'posts',
+        method: 'POST',
+        body,
+      }),
+    }),
+    editPost: build.mutation({
+      query: (body) => ({
+        url: `posts/${body.id}`,
+        method: 'POST',
+        body,
+      }),
+    }),
+  }),
+})
+```
+
+For more granular control over the provided data, provided `tags` can have an associated `id`. This enables a distinction between 'any of a particular tag type', and 'a specific instance of a particular tag type'.
+
+The example below declares that the provided posts are associated with particular IDs as determined by the result returned by the endpoint:
+
+```js title="Example of providing tags with IDs to the cache"
+const api = createApi({
+  baseQuery,
+  tagTypes: ['Post', 'User'],
+  endpoints: (build) => ({
+    getPosts: build.query({
+      query: () => '/posts',
+      // highlight-start
+      providesTags: (result, error, arg) =>
+        result
+          ? [...result.map(({ id }) => ({ type: 'Post', id })), 'Post']
+          : ['Post'],
+      // highlight-end
+    }),
+    getUsers: build.query({
+      query: () => '/users',
+      providesTags: ['User'],
+    }),
+    addPost: build.mutation({
+      query: (body) => ({
+        url: 'posts',
+        method: 'POST',
+        body,
+      }),
+    }),
+    editPost: build.mutation({
+      query: (body) => ({
+        url: `posts/${body.id}`,
+        method: 'POST',
+        body,
+      }),
+    }),
+  }),
+})
+```
+
+Note that for the example above, the `id` is used where possible on a successful result. In the case of an error, no result is supplied, and we still consider that it has provided the general `'Post'` tag type rather than any specific instance of that tag.
```

Comment:
Hmm. I'm not 100% sure if providing the general `['Post']` tag makes any sense here. That would invalidate only if an endpoint would invalidate *all* posts by `invalidatesTags: ['Post']`.
I'd rather have a `LIST` id in both results as well.

Reply:
PS: reading further I why it is still missing here though

Reply:
Yeah the goal was to try to introduce it as a general concept piece by piece first rather than immediately going into 'This is how you should be using it'.

I've added another 'tip' block in the 'Providing cache data' section linking to the 'Advanced Invalidation' section which hopefully is clear enough.